### PR TITLE
:bug: Fix the HTML converter when collection ends in failure

### PR DESCRIPTION
**Pull request by ouvreboite**
When a collection's query ends in failure, the corresponding run.execution does not contains a "response" object.

So we should only recreate a Response only if there si one.

Otherwise the converter try to access its "stream" prop (which is undefined in our case).

### CODE REVIEW

File: src/app.js [hunk 1]
```diff
@@ -151,6 +153,8 @@ class Application {
             `An error occured while converting JSON summary to HTML report.`,
             error
           );
+          logger.error(error?.stack);
```
Comment (PaulMathon) [inline src/app.js:156]: I would update the logger.error method so that the stack in printed

File: tests/resources/summary-failure.json [hunk 2]
```diff
@@ -0,0 +1,543 @@
...
+                  "ref": "2e037369-8fde-4283-96a0-98381325a6a1",
+                  "httpRequestId": "2fa25e3f-6599-461c-b129-e48098f1c377",
+                  "scriptId": "8341984b-78b9-43bd-8120-6ba20de8ac8c",
+                  "execution": "f6f7eaf8-b2d6-440d-b68a-415ae0e82965"
+              }
+          }
+      ],
+      "error": null
+  }
+}
```
Comment (PaulMathon) [inline tests/resources/summary-failure.json:543]: OPT: add line

File: src/api/convert-html.js [hunk 3]
```diff
@@ -6,10 +6,12 @@ function generateHTMLReport(summary) {
...
-    exec.item = new Item(exec.item);
-    exec.item.setParent(collection);
-    exec.request = new Request(exec.request);
-    exec.response = new Response(exec.response);
+    exec.item = exec.item ? new Item(exec.item) : undefined;
+    if(exec.item)
+      exec.item.setParent(collection);
+    exec.request = exec.request ? new Request(exec.request) : undefined;
+    exec.response = exec.response ? new Response(exec.response) : undefined;
+    exec.failure = Failure()
```
Comment (PaulMathon) [inline src/api/convert-html.js:14]: Where does the object 'Failure' is coming from?

File: tests/resources/summary-failure.json (hunk 2 above)
Comment (ouvreboite) [reply tests/resources/summary-failure.json:543]: done

File: src/app.js (hunk 1 above)
Comment (ouvreboite) [reply src/app.js:156]: done

**Review by PaulMathon — APPROVED** [review]

**Pull request merged**